command/operator_raft_snapshot_save.go

```diff
@@ -71,24 +72,48 @@ func (c *OperatorRaftSnapshotSaveCommand) Run(args []string) int {
 		return 1
 	}
 
-	snapFile, err := os.OpenFile(path, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0644)
-	if err != nil {
-		c.UI.Error(fmt.Sprintf("Error opening output file: %s", err))
-		return 2
+	w := &lazyOpenWriter{
+		openFunc: func() (io.WriteCloser, error) {
+			return os.OpenFile(path, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0644)
+		},
 	}
-	defer snapFile.Close()
+	defer w.Close()
```

Comment:
I know this was in the original code, but isn't this a case where we really should be checking the result of the `Close` call?

Reply:
Sure?  It makes the code a little uglier but what the heck.